src/pudl_archiver/archivers/ferc/ferc1.py

```diff
@@ -23,8 +23,21 @@ async def get_resources(self) -> ArchiveAwaitable:
         filings = xbrl.index_available_entries()[xbrl.FercForm.FORM_1]
         for year, year_filings in filings.items():
             if self.valid_year(year):
+                if year > 2019:
+                    yield self.get_year_taxonomy_xbrl(year)
                 yield self.get_year_xbrl(year, year_filings)
 
+    async def get_year_taxonomy_xbrl(self, year: int) -> tuple[Path, dict]:
```

Comment:
Could be nice to create a base XBRL archiver class that each of these can inherit from or have as a class member to avoid this duplicated logic.

Reply:
Yeah, that would be nice :) Since the "getting a taxonomy" functionality isn't particularly form-specific, I would probably shunt that into a helper function that's not in any class, `ferc.xbrl.get_taxonomy(year, form, download_dir, session)` - and then include the download_dir and session at the callsite.

Reply:
Good call, I was definitely overthinking this. I've just changed `archive_taxonomy` to return a `ResourceInfo` object.